lib/install.js

```diff
@@ -215,23 +217,31 @@ function Installer (where, dryrun, args) {
   this.progress = {}
   this.noPackageJsonOk = !!args.length
   this.topLevelLifecycles = !args.length
-  this.dev = npm.config.get('dev') || (!/^prod(uction)?$/.test(npm.config.get('only')) && !npm.config.get('production')) || /^dev(elopment)?$/.test(npm.config.get('only'))
-  this.prod = !/^dev(elopment)?$/.test(npm.config.get('only'))
-  this.rollback = npm.config.get('rollback')
-  this.link = npm.config.get('link')
-  this.global = this.where === path.resolve(npm.globalDir, '..')
+  this.dev = opts.dev != null ? opts.dev : npm.config.get('dev') || (!/^prod(uction)?$/.test(npm.config.get('only')) && !npm.config.get('production')) || /^dev(elopment)?$/.test(npm.config.get('only'))
```

Comment:
Can I ask that we parenthesize, and maybe split between lines, all these different conditionals? I can never tell how it's actually supposed to parse when you do ternary + or + and all in on line >_<

Reply:
this change otherwise looks good

Reply:
er wow, yeah, that looks awful…